### ISSUE TITLE
'behavior' Parameter Passed In The Wrong Order in Regex Builder Refactoring

### ISSUE THREAD

**Issue by CodaFi** [apple/swift-experimental-string-processing#608]
In Xcode 14.1 beta 3, this regex

```swift
#/(.*?)/#
```

Expands to this regex builder

```swift
Regex {
  Capture {
    ZeroOrMore(.reluctant, .any)
  }
}
```

Which fails to type check because `ZeroOrMore` takes the component first and the behavior second when not passed a trailing closure that produces a component.

**Comment by milseman** [comment]
CC @DaveEwing

**Comment by Azoy** [comment]
On main this get's printed as:
```swift
Regex {
  Capture {
    ZeroOrMore(.reluctant) {
      /./
    }
  }
}
```
with the comment that the DSL'S `.any` is not equivalent to `/./`. cc: @hamishknight

**Comment by hamishknight** [comment]
Yeah the `.` of a regex literal has different newline matching behavior to `.any`. However looks like this is still an issue for other elements, e.g on main `/\d*?/` gets refactored to:

```swift
Regex {
  ZeroOrMore(.reluctant, .digit)
}
```
